src/ui/public/vis/editors/config/editor_config_providers.ts

```diff
@@ -0,0 +1,133 @@
+/*
+ * Licensed to Elasticsearch B.V. under one or more contributor
+ * license agreements. See the NOTICE file distributed with
+ * this work for additional information regarding copyright
+ * ownership. Elasticsearch B.V. licenses this file to you under
+ * the Apache License, Version 2.0 (the "License"); you may
+ * not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *    http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing,
+ * software distributed under the License is distributed on an
+ * "AS IS" BASIS, WITHOUT WARRANTIES OR CONDITIONS OF ANY
+ * KIND, either express or implied.  See the License for the
+ * specific language governing permissions and limitations
+ * under the License.
+ */
+
+import { AggConfig } from '../..';
+import { AggType } from '../../../agg_types';
+import { IndexPattern } from '../../../index_patterns';
+import { EditorConfig, EditorParamConfig, FixedParam, NumericIntervalParam } from './types';
+
+type EditorConfigProvider = (
+  aggType: AggType,
+  indexPattern: IndexPattern,
+  aggConfig: AggConfig
+) => EditorConfig;
+
+function greatestCommonDivisor(a: number, b: number): number {
```

Comment:
@jen-huang Weren't you looking for utilities like this?

These seem general enough to belong in their own `math` module in `ui/public`, with a couple unit tests. What do you think @timroes?

Reply:
Yeap, I will move them